src/main/java/com/guysfromusa/carsgame/control/GameEngine.java

```diff
@@ -61,15 +62,15 @@ public void handleMoves(List<Command> commands, String gameId) {
     public void handleAddCars(List<Command> commands, String gameName) {
         GameState gameState = activeGamesContainer.getGameState(gameName);
 
-        commands.forEach(command -> {
-            AddCarToGameCommand cmd = (AddCarToGameCommand) command;
-            CompletableFuture<CarEntity> result = cmd.getFuture();
-            CarEntity carEntity = carService.addCarToGame(cmd.getCarName(), gameName, cmd.getStartingPoint());
-            result.complete(carEntity);
+        commands.stream()
+                .map(command -> (AddCarToGameCommand) command)
+                .map(cmd -> Tuple.of(cmd, Try.of(() -> carService.addCarToGame(cmd.getCarName(), gameName, cmd.getStartingPoint())).toEither()))
+                .map(cmdEitherTuple2 -> cmdEitherTuple2._2.isRight()
+                        ? (Executable) () -> cmdEitherTuple2._1.getFuture().complete(cmdEitherTuple2._2.get())
+                        : (Executable) () -> cmdEitherTuple2._1.getFuture().completeExceptionally(cmdEitherTuple2._2.getLeft()))
+                .forEach(Executable::invoke);
```

Comment:
is ti the same:

    commands.stream()
                .map(command -> (AddCarToGameCommand) command)
                .map(cmd -> Tuple.of(cmd.getFuture(), 
                        Try.of(() -> carService.addCarToGame(cmd.getCarName(), gameName, cmd.getStartingPoint()))
                        .toEither())
                )
                .forEach(futureAndEither -> futureAndEither._2
                        .mapLeft(futureAndEither._1::completeExceptionally)
                        .right()
                        .forEach(futureAndEither._1::complete));
?

Reply:
instead of executable isnt it better to replace last map with foreach?